plugin/evm/contracts_test.go

```diff
@@ -0,0 +1,133 @@
+)
+
+var (
+	testAbiStore            abi.ABI
+	testAbiVotePower        abi.ABI
+	testAbiWnat             abi.ABI
+	testAbiReward           abi.ABI
+	testAbiManager          abi.ABI
+	testAbiFtsoRegistry     abi.ABI
+	testAbiVoterWhitelister abi.ABI
+	testAbiValidation       abi.ABI
+	testAbiSubmitter        abi.ABI
+)
+
+func initTestContracts(t *testing.T) {
+	t.Helper()
+
+	var err error
+
+	testAbiStore, err = abi.JSON(strings.NewReader(testStoreJSON))
+	require.NoError(t, err)
+
+	testAbiVotePower, err = abi.JSON(strings.NewReader(testAbiVotePowerJSON))
+	require.NoError(t, err)
+
+	testAbiWnat, err = abi.JSON(strings.NewReader(testAbiWnatJSON))
+	require.NoError(t, err)
+
+	testAbiReward, err = abi.JSON(strings.NewReader(testAbiRewardJSON))
+	require.NoError(t, err)
+
+	testAbiManager, err = abi.JSON(strings.NewReader(testAbiManagerJSON))
+	require.NoError(t, err)
+
+	testAbiFtsoRegistry, err = abi.JSON(strings.NewReader(testAbiFtsoRegistryJSON))
+	require.NoError(t, err)
+
+	testAbiVoterWhitelister, err = abi.JSON(strings.NewReader(testAbiVoterWhitelisterJSON))
+	require.NoError(t, err)
+
+	testAbiValidation, err = abi.JSON(strings.NewReader(testAbiValidationJSON))
+	require.NoError(t, err)
+
+	testAbiSubmitter, err = abi.JSON(strings.NewReader(testAbiSubmitterJSON))
+	require.NoError(t, err)
+}
+
+func deployTestContract(t *testing.T, auth *bind.TransactOpts, be *backends.SimulatedBackend, testAbi abi.ABI, testAbiBin string, params ...interface{}) common.Address {
+	t.Helper()
+
+	addr, _, _, err := bind.DeployContract(auth, testAbi, common.FromHex(testAbiBin), be, params...)
+	require.NoError(t, err)
+
+	return addr
+}
+
+func simulatedBlockchain(t *testing.T) (*bind.TransactOpts, *backends.SimulatedBackend) {
+	t.Helper()
+
+	key, err := crypto.GenerateKey()
+	require.NoError(t, err)
+
+	auth, err := bind.NewKeyedTransactorWithChainID(key, big.NewInt(1337))
+	require.NoError(t, err)
+
+	balance := big.NewInt(0).SetUint64(math.MaxUint64)
+	alloc := make(core.GenesisAlloc)
+	alloc[auth.From] = core.GenesisAccount{
+		Balance: balance,
+	}
+	gasLimit := uint64(math.MaxUint64)
```

Comment:
No, it's not. It's an integer constant that will be interpreted as an `int` if no specific type is given to the variable (i.e. if type inference is used there). And then you get a compilation error because the value doesn't fit into an integer.